### ISSUE TITLE
Eclipse Kura can't access additional UARTs in Raspberry Pi 4

### ISSUE THREAD

**Issue by alexcorvis84** [eclipse/kura#4360]
**Describe the bug**

- Kura can't access additional UARTs in Raspberry Pi 4.

**To Reproduce & Expected behavior**

I'm trying to read a sensor from Serial in a **Raspberry Pi 4** running **KURA_5.2.0**. I've installed the [Serial Example](https://github.com/eclipse/kura/tree/develop/kura/examples/org.eclipse.kura.example.serial.publisher) but I'm receiving the following error when trying to open de port `/dev/ttyAMA1`.

I've checked that if I setup `/dev/ttyAMA0 `(which is the [primary UART0](https://github.com/raspberrypi/documentation/blob/develop/documentation/asciidoc/computers/configuration/uart.adoc#primary-uart) present on GPIO 14 (TX) - 15 (RX), port is opened OK, but I need to get it working on /dev/ttyAMA1 (which is [UART2](https://github.com/raspberrypi/documentation/blob/develop/documentation/asciidoc/computers/configuration/uart.adoc#uarts-and-device-tree) present on GPIO 0 (TX) - 1 (RX).

`/dev/ttyAMA0` **Working** ✅😃
![imagen](https://user-images.githubusercontent.com/44984326/216545767-73d4f5c8-5b84-4f79-a1b7-421606c1c559.png)

`/dev/ttyAMA1` **Failed to open port** ❌😔
![imagen](https://user-images.githubusercontent.com/44984326/216545823-b68a3ab7-956e-4530-84b5-ef2f2303da3f.png)

**Target Environment (please complete the following information):**
 - Board: **Raspberry Pi 4**
 - OS version: Linux raspberrypi 5.15.76-v7l+ 1597 SMP Fri Nov 4 12:14:58 GMT 2022 armv7l GNU/Linux
 - `cat /etc/os-release` output

```
PRETTY_NAME="Raspbian GNU/Linux 11 (bullseye)"
NAME="Raspbian GNU/Linux"
VERSION_ID="11"
VERSION="11 (bullseye)"
VERSION_CODENAME=bullseye
ID=raspbian
ID_LIKE=debian
HOME_URL="http://www.raspbian.org/"
SUPPORT_URL="http://www.raspbian.org/RaspbianForums"
BUG_REPORT_URL="http://www.raspbian.org/RaspbianBugs"
```
 - Additional info: **KURA_5.2.0**

**Additional context**

I've developed a Serial Driver plugin based on [Serial Example](https://github.com/eclipse/kura/tree/develop/kura/examples/org.eclipse.kura.example.serial.publisher). I can confirm it's working as I can read the data **ONLY** if I connect the sensor on port `/dev/ttyAMA0`. But as the Raspberry Pi 4 lets you use 4 additional UART ports, I want _and neeed_ to add more serial sensors on those additional UARTs in order to read the data from Kura.

Example of a data read CO2 sensor working on /dev/ttyAMA0:

![imagen](https://user-images.githubusercontent.com/44984326/216547661-aae2539b-03e5-4716-a9d3-943524ab0a43.png)

![imagen](https://user-images.githubusercontent.com/44984326/216547757-9bea3c7f-7587-48f0-abc5-5e15e84378ee.png)

![imagen](https://user-images.githubusercontent.com/44984326/216547848-9dffe1a7-67af-4f21-9beb-e1c41a109abf.png)

**Comment by salvatore-coppola** [comment]
Hi @alexcorvis84 , from the log screenshot you uploaded in the failing scenario I noticed that the publisher try to open the `/dev/ttyAMA2` in place of `/dev/ttyAMA1`. 
Howevery you could upload the snapshot of the failing scenario so we can check the example publisher configuration.

Regards.

**Comment by alexcorvis84** [comment]
> Hi @alexcorvis84 , from the log screenshot you uploaded in the failing scenario I noticed that the publisher try to open the `/dev/ttyAMA2` in place of `/dev/ttyAMA1`. However you could upload the snapshot of the failing scenario so we can check the example publisher configuration.
> 
> Regards.

Hi @salvatore-coppola,

First of all, thank you for your quick reply!

Yeah, sorry for mixing you up. In the comments I indicated `/dev/ttyAMA1` whereas the screenshots shows `/dev/ttyAMA2`, but it does not matter. I mean, **Kura** is **only able** to get working the primary UART and not the 4 additional UART ports that Raspberry Pi 4 lets you configure and use. In the scenario explained (by the screenshots) I was trying to open `/dev/ttyAMA2` and **it didn't work** either 😕

As requested, below 📎 attached the snapshot.

📎[snapshot_1675668503453.zip](https://github.com/eclipse/kura/files/10660668/snapshot_1675668503453.zip)

Meanwhile, I've checked that Serial connection relies upon **_javax.comm_** class. 

In order to check & list the ports that are recognized by Kura, I coded below piece of code and it only shows the 'default' ones (hope it can help you to address what's happening... 🤔)

```
        Enumeration<?> e = CommPortIdentifier.getPortIdentifiers();

        while ( e.hasMoreElements() ) 
        {
            CommPortIdentifier portIdentifier = (CommPortIdentifier) e.nextElement();
            logger.info("Port Name: {} - Port Type: {}", portIdentifier.getName()  , getPortTypeName(portIdentifier.getPortType()) );
        }
```

![imagen](https://user-images.githubusercontent.com/44984326/216913094-e59a77c2-fbf5-41e1-a568-460531bac8f9.png)

Please let me know any findings that could help me to address to use the added UART ports.

**Comment by salvatore-coppola** [comment]
Hi @alexcorvis84, just a quick question , have you enabled the additional uart ports on the raspberry Pi and created the corresponding linux serial interfaces?

**Comment by salvatore-coppola** [comment]
Hi @alexcorvis84 any news on that ? Just to see if this requires further investigation on our part.

**Comment by alexcorvis84** [comment]
> Hi @alexcorvis84, just a quick question , have you enabled the additional uart ports on the raspberry Pi and created the corresponding linux serial interfaces?

Hi @salvatore-coppola , first of all, sorry about being late replying.

Yes, I enabled them and created the linux serial interfaces.

Finally, I **solved** the problem:

1. Stopping and starting Kura once I enabled the additional ports (if not, it didn't acknowledge the changes)
2. I needed to create symbolic links as detailed [on this post](https://www.eclipse.org/forums/index.php?t=msg&th=905751&goto=1524867&#msg_1524867).

With this two changes, now I can use the extra UART ports in Kura without problems :)

Thank you for your support.
